File: leetCode/two-pointers/language/go/is-palyndrome.go
package main

import "fmt"

func main() {
	fmt.Println(isPalindrome("abba"))
	fmt.Println(isPalindrome("abbaa"))
}

func isPalindrome(s string) bool {
	if len(s) == 0 {
		return true
	}

	left := 0
	right := len(s) - 1

	for left < right {
		if s[left] != s[right] {
			return false
		}

		left++
		right--
	}

	return true
}

File: leetCode/two-pointers/language/rust/is-palyndrome.rs
fn isPalyndrome(s: &str) -> bool {
    let mut left = 0;
    let mut right = s.len() - 1;
    while left < right {
        let leftChar = s.chars().nth(left).unwrap();
        let rightChar = s.chars().nth(right).unwrap();
        if leftChar != rightChar {
            return false;
        }
        left += 1;
        right -= 1;
    }
    true
}

fn main() {
    let s = "abba";
    let result = isPalyndrome(s);
    println!("result: {}", result);
}
